sql/src/main/java/io/crate/executor/transport/task/elasticsearch/ESGetTask.java

```diff
@@ -22,196 +22,144 @@
-            return tableInfo.ident().indexName();
+        @Override
+        protected void innerStart() {
+            transportAction.execute(request, this);
         }
     }
 
-    private GetRequest prepareGetRequest(ESGet node, FetchSourceContext fsc) {
-        DocKeys.DocKey docKey = node.docKeys().getOnlyKey();
-        GetRequest getRequest = new GetRequest(indexName(node.tableInfo(), docKey.partitionValues()),
-                Constants.DEFAULT_MAPPING_TYPE, docKey.id());
-        getRequest.fetchSourceContext(fsc);
-        getRequest.realtime(true);
-        getRequest.routing(docKey.routing());
-        return getRequest;
-    }
+    private static class MultiGetJobContext extends JobContext<TransportMultiGetAction, MultiGetRequest, MultiGetResponse> {
+
 
-    private MultiGetRequest prepareMultiGetRequest(ESGet node, FetchSourceContext fsc) {
-        MultiGetRequest multiGetRequest = new MultiGetRequest();
-        for (DocKeys.DocKey key : node.docKeys()) {
-            MultiGetRequest.Item item = new MultiGetRequest.Item(
-                    indexName(node.tableInfo(), key.partitionValues()), Constants.DEFAULT_MAPPING_TYPE, key.id());
-            item.fetchSourceContext(fsc);
-            item.routing(key.routing());
-            multiGetRequest.add(item);
+        MultiGetJobContext(ESGetTask task,
+                                  TransportMultiGetAction transportAction,
+                                  RowReceiver downstream) {
+            super(task, transportAction, downstream);
+            assert task.esGet.docKeys().size() > 1;
+            assert task.projectorFactory != null;
         }
-        multiGetRequest.realtime(true);
-        return multiGetRequest;
-    }
 
-    private FlatProjectorChain getFlatProjectorChain(ProjectorFactory projectorFactory,
-                                                     ESGet node,
-                                                     QueryResultRowDownstream queryResultRowDownstream) {
-        if (node.limit() > TopN.NO_LIMIT || node.offset() > 0 || !node.sortSymbols().isEmpty()) {
-            List<Symbol> orderBySymbols = new ArrayList<>(node.sortSymbols().size());
-            for (Symbol symbol : node.sortSymbols()) {
-                int i = node.outputs().indexOf(symbol);
-                if (i < 0) {
-                    orderBySymbols.add(new InputColumn(node.outputs().size() + orderBySymbols.size()));
-                } else {
-                    orderBySymbols.add(new InputColumn(i));
+        @Override
+        protected void innerPrepare() throws Exception {
+            FlatProjectorChain projectorChain = getFlatProjectorChain(downstream);
+            downstream = projectorChain.firstProjector();
+            projectorChain.prepare();
+        }
+
+        private FlatProjectorChain getFlatProjectorChain(RowReceiver downstream) {
+            if (task.esGet.limit() > TopN.NO_LIMIT || task.esGet.offset() > 0 || !task.esGet.sortSymbols().isEmpty()) {
```

Comment:
afaik checking for offset here is not needed as limit must always be set if offset is set